### DISCUSSION TITLE
Highlight code doesn't work

### DISCUSSION THREAD

**Discussion by Flayzera** [vuejs/vitepress#3745] [Q&A] [answered]
When I try to embed code into a block code that is coming from frontmatter it doesn't highlight the code.

Example:

``````md
---
somecode: |
  {
    foo: bar
  }
---

```json-vue
{{ $frontmatter.somecode }}
```

``````

**Comment by brc-dd** [comment, marked as answer]
It's expected. We can't fix it. Refer the last line here - https://vitepress.dev/guide/using-vue#unescape-in-code-blocks 

You'll need to use data loader to generate html from that and insert that using v-html if you want syntax highlighting there.

**Reply by Vanuan** [reply]
What's the point of specifying the language then? Could've just used 
```<pre>{{ $frontmatter.somecode }}</pre>``` with the same result

**Reply by brc-dd** [reply]
@Vanuan Directly using pre won’t add some of the stuff like copy code button or language label and related styling, won’t work with code groups either.

**Reply by Vanuan** [reply]
Got it, thanks
